library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -334,21 +477,32 @@ private String getDisplayCaption() {
       return null;
     }
 
+    // Every styling span should be closed at end of line
+    closeOpenSpans( STYLE_PRIORITY.STYLE_PRIORITY_ALL );
```

Comment:
This doesn't look right. There's no guarantee that a line has ended when getDisplayCaption is called. In particular, this would not be true when in paint-on caption mode.

I think it you might need to do something like adding spans up-front using SPAN_EXCLUSIVE_INCLUSIVE, then replacing them SPAN_EXCLUSIVE_EXCLUSIVE spans when you actually reach an end-of-line.